content/en/docs/ops/common-problems/security-issues/index.md

```diff
@@ -408,3 +408,82 @@ second ensure that [keys and certificates are being delivered](#keys-and-certifi
+as expected, please first read the
+[documentation](/docs/tasks/security/authentication/authn-policy/#auto-mutual-tls).
+
+Auto mTLS has a known limitation with [workload level](/docs/tasks/security/authentication/authn-policy/#enable-mutual-tls-per-workload) peer authentication.
+When workload level peer authentication is used, you need to configure a corresponding
+destination rule configuration to let client sidecar send mTLS or plaintext
+traffic correctly.
+
+If you still think there's an issue, follow the instructions below for
+investigation.
+
+1. Get the client envoy cluster configuration to ensure auto mTLS is configured.
+You should see `transportSocketMatches` configured for the given Envoy cluster.
+
+    {{< text bash >}}
+    $ POD=$(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..metadata.name})
+    $ istioctl proxy-config clusters ${POD}.foo -ojson | grep 'name.*outbound.*httpbin' -A130
+    "name": "outbound|8000||httpbin.default.svc.cluster.local",
+    ...
+    "transportSocketMatches": [
+            {
+                "name": "tlsMode-istio",
+                "match": {
+                    "tlsMode": "istio"
+                },
+                "transportSocket": {
+                    "name": "envoy.transport_sockets.tls",
+                    "typedConfig": {
+                        "@type": "type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext",
+                        "commonTlsContext": ...
+            }
+            {
+                "name": "tlsMode-disabled",
+                "match": {},
+                "transportSocket": {
+                    "name": "envoy.transport_sockets.raw_buffer"
+                }
+            }
+    {{< /text >}}
+
+1. Check the destination pod to see if the destination endpoint has the label
+`tlsMode` equals to `istio`.
+
+    {{< text bash >}}
+    $ POD=$(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..metadata.name})
+    $ IP=$(kubectl get pod -lapp=sleep -o jsonpath="{.items[*].status.podIP}")
+    $ kubectl exec -it ${POD} -c istio-proxy -- curl 'localhost:15000/config_dump?include_eds=true' | grep ${IP} -A15 -B5
+      "endpoint": {
+        "address": {
+        "socket_address": {
+            "address": "10.32.28.136",
+            "port_value": 80
+        }
+        ...
+        "metadata": {
+          "filter_metadata": {
+            "envoy.transport_socket_match": {
+              "tlsMode": "istio"
+            }
+        },
+    {{< /text >}}
+
+1. Check Envoy upstream debug log to confirm what transport socket configuration is being
+used. This requires to change Envoy upstream log to debug level.
+
+    {{< text bash >}}
+    $ POD=$(kubectl get pod -l app=sleep -o jsonpath={.items..metadata.name})
+    $ istioctl pc log sleep-8f795f47d-rfhdt --level debug
+    2021-04-08T22:35:22.650478Z     debug   envoy upstream  transport socket match, socket tlsMode-istio selected for host with address 10.32.28.136:80
```

Comment:
this is the actual log and not the output of the istioctl command? I think we should include the specific command for getting the log.

Reply:
done.